calico/felix/fetcd.py

```diff
@@ -456,8 +461,45 @@ def _build_config_dict(cfg_node):
 
 
 # Intern JSON keys as we load them to reduce occupancy.
+FIELDS_TO_INTERN = set([
+    # Endpoint dicts.  It doesn't seem worth interning items like the MAC
+    # address or TAP name, which are rarely (if ever) shared.
+    "profile_id",
+    "profile_ids",
+    "state",
+    "ipv4_gateway",
+    "ipv6_gateway",
+
+    # Rules dicts.
+    "protocol",
+    "src_tag",
+    "dst_tag",
+    "action",
+])
 def intern_dict(d):
-    return dict((intern(str(k)), v) for k, v in d.iteritems())
+    out = {}
```

Comment:
This is a good utility function - better to put it in either `futils.py` or even `common.py`.